src/ConsoleArgumentBag.php

```diff
@@ -0,0 +1,113 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Tempest\Console;
+
+use Tempest\Console\Exceptions\UnresolvedArgumentsException;
+
+final class ConsoleArgumentBag
+{
+    /** @var ConsoleInputArgument[] */
+    protected array $arguments = [];
+
+    /** @var string[] */
+    protected array $path = [];
+
+    /**
+     * @param array<string|int, mixed> $arguments
+     */
+    public function __construct(array $arguments)
+    {
+        $this->path = array_filter([
+            $arguments[0] ?? null,
+            $arguments[1] ?? null,
+        ]);
+
+        unset($arguments[0], $arguments[1]);
+
+        foreach (array_values($arguments) as $position => $argument) {
+            if (str_starts_with($argument, '--')) {
+                $parts = explode('=', str_replace('--', '', $argument));
+
+                $key = $parts[0];
+
+                $this->set($key, $parts[1] ?? true);
+
+                continue;
+            }
+
+            $this->set($position, $argument);
+        }
+    }
+
+    /**
+     * @return ConsoleInputArgument[]
+     */
+    public function all(): array
+    {
+        return $this->arguments;
+    }
+
+    public function get(string|int $name): ?ConsoleInputArgument
+    {
+        return $this->arguments[$name] ?? null;
+    }
+
+    public function has(string $name): bool
+    {
+        return array_key_exists($name, $this->arguments);
+    }
+
+    public function set(string|int $name, mixed $value): ConsoleInputArgument
+    {
+        return $this->arguments[$name] = new ConsoleInputArgument(
+            $name,
+            $value,
+            $value,
+            position: count($this->arguments),
+        );
+    }
+
+    public function resolveInput(ConsoleCommand $command): ConsoleCommandInput
```

Comment:
I'm confused by this function. In my mind an argument bag is created from `$_SERVER['argv']`, then that argument bag is validated against a console command, and possible attached to the console command.

Why then would the argument bag need this method?

Reply:
I can see your point - how else would you want to approach this? Keep the resolving in ConsoleApplication like it used to be?